dataRetriever/storageResolvers/trieNodeResolver_test.go

```diff
@@ -1,3 +1,158 @@
+		ResponseTopicName:        "",
+		Marshalizer:              &mock.MarshalizerStub{},
+		TrieDataGetter:           &mock.TrieStub{},
+		TrieStorageManager:       &mock.TrieStorageManagerStub{},
+		ManualEpochStartNotifier: &mock.ManualEpochStartNotifierStub{},
+		ChanGracefullyClose:      make(chan endProcess.ArgEndProcess, 1),
+		DelayBeforeGracefulClose: 0,
+	}
+}
+
+func TestNewTrieNodeResolver_InvalidArgumentsShouldErr(t *testing.T) {
+	t.Parallel()
+
+	args := createMockTrieResolverArguments()
+	args.Messenger = nil
+	tnr, err := NewTrieNodeResolver(args)
+	assert.True(t, check.IfNil(tnr))
+	assert.Equal(t, dataRetriever.ErrNilMessenger, err)
+
+	args = createMockTrieResolverArguments()
+	args.ManualEpochStartNotifier = nil
+	tnr, err = NewTrieNodeResolver(args)
+	assert.True(t, check.IfNil(tnr))
+	assert.Equal(t, dataRetriever.ErrNilManualEpochStartNotifier, err)
+
+	args = createMockTrieResolverArguments()
+	args.ChanGracefullyClose = nil
+	tnr, err = NewTrieNodeResolver(args)
+	assert.True(t, check.IfNil(tnr))
+	assert.Equal(t, dataRetriever.ErrNilGracefullyCloseChannel, err)
+
+	args = createMockTrieResolverArguments()
+	args.TrieStorageManager = nil
+	tnr, err = NewTrieNodeResolver(args)
+	assert.True(t, check.IfNil(tnr))
+	assert.Equal(t, dataRetriever.ErrNilTrieStorageManager, err)
+
+	args = createMockTrieResolverArguments()
+	args.TrieDataGetter = nil
+	tnr, err = NewTrieNodeResolver(args)
+	assert.True(t, check.IfNil(tnr))
+	assert.Equal(t, dataRetriever.ErrNilTrieDataGetter, err)
+
+	args = createMockTrieResolverArguments()
+	args.Marshalizer = nil
+	tnr, err = NewTrieNodeResolver(args)
+	assert.True(t, check.IfNil(tnr))
+	assert.Equal(t, dataRetriever.ErrNilMarshalizer, err)
+}
+
+func TestNewTrieNodeResolver_ShouldWork(t *testing.T) {
+	t.Parallel()
+
+	args := createMockTrieResolverArguments()
+	tnr, err := NewTrieNodeResolver(args)
+	assert.False(t, check.IfNil(tnr))
+	assert.Nil(t, err)
+}
+
+func TestTrieNodeResolver_RequestDataFromHashGetSubtrieFailsShouldErr(t *testing.T) {
+	t.Parallel()
+
+	args := createMockTrieResolverArguments()
+	expectedErr := errors.New("expected error")
+	args.TrieDataGetter = &mock.TrieStub{
+		GetSerializedNodesCalled: func(bytes []byte, u uint64) ([][]byte, uint64, error) {
+			return nil, 0, expectedErr
+		},
+	}
+	tnr, _ := NewTrieNodeResolver(args)
+
+	err := tnr.RequestDataFromHash(nil, 0)
+	assert.Equal(t, expectedErr, err)
+
+	select {
+	case <-args.ChanGracefullyClose:
+	case <-time.After(time.Second):
+		assert.Fail(t, "timout while waiting to signal on gracefully close channel")
+	}
+}
+
+func TestTrieNodeResolver_RequestDataFromHashShouldWork(t *testing.T) {
+	t.Parallel()
+
+	args := createMockTrieResolverArguments()
+	buff := []byte("data")
+	args.TrieDataGetter = &mock.TrieStub{
+		GetSerializedNodesCalled: func(bytes []byte, u uint64) ([][]byte, uint64, error) {
+			return [][]byte{buff}, 1, nil
+		},
+	}
+	numSendCalled := uint32(0)
```

Comment:
rename to `countSendToConnectedPeerCalled`

Reply:
`numSendToConnectedPeerCalled` as to be consistent across other test files